From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Glitch <glitch.g3431@gmail.com>
Date: Sat, 12 Mar 2022 21:08:03 -0600
Subject: [PATCH] Rebrand to Quilt

This includes:
- Changing the plugin marker, including in the tests
- Using a different set of caches
- Using the Quilt Loader system properties
- Using `quilt_installer.json`
- Renaming the shared service manager
- Automatically applying the Quilt Maven
- Adding Vineflower and using it by default

diff --git a/bootstrap/test-project/build.gradle b/bootstrap/test-project/build.gradle
index 561915428a80131439ea9e1d9f12804c491d399a..db648181f772520d1607ce2510a66be1d00167c2 100644
--- a/bootstrap/test-project/build.gradle
+++ b/bootstrap/test-project/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.13.local'
+	id 'org.quiltmc.loom' version '0.13.local'
 }
 
 dependencies {
diff --git a/build.gradle b/build.gradle
index c7ff152af65d5e2724df765f3539de2df0a1d519..6ee2dea93e8453387cfb996f72e5c8b7457a0fd1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -23,7 +23,7 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
 	}
 }
 
-group = 'net.fabricmc'
+group = 'org.quiltmc'
 archivesBaseName = project.name
 def baseVersion = '1.3'
 
@@ -45,6 +45,11 @@ repositories {
 		name = 'Fabric'
 		url = 'https://maven.fabricmc.net/'
 	}
+	maven {
+		name = "Quilt"
+		url = "https://maven.quiltmc.org/repository/release"
+	}
+	gradlePluginPortal()
 	mavenCentral()
 }
 
@@ -108,6 +113,11 @@ dependencies {
 	// Kapt integration
 	compileOnly libs.kotlin.gradle.plugin
 
+	// Plugins we apply
+	implementation('io.github.juuxel:loom-vineflower:1.11.0') {
+		exclude group: 'net.fabricmc', module: 'fabric-loom'
+	}
+
 	// Testing
 	testImplementation(gradleTestKit())
 	testImplementation('org.spockframework:spock-core:2.3-groovy-3.0') {
@@ -184,8 +194,8 @@ codenarc {
 
 gradlePlugin {
 	plugins {
-		fabricLoom {
-			id = 'fabric-loom'
+		quiltLoom {
+			id = 'org.quiltmc.loom'
 			implementationClass = 'net.fabricmc.loom.bootstrap.LoomGradlePluginBootstrap'
 		}
 	}
@@ -242,8 +252,8 @@ publishing {
 
 		// Manually crate the plugin marker for snapshot versions
 		snapshotPlugin(MavenPublication) { publication ->
-			groupId 'fabric-loom'
-			artifactId 'fabric-loom.gradle.plugin'
+			groupId 'org.quiltmc.loom'
+			artifactId 'org.quiltmc.loom.gradle.plugin'
 			version baseVersion + '-SNAPSHOT'
 
 			pom.withXml({
@@ -253,9 +263,9 @@ publishing {
 				Node dependencies = root.appendChild(document.createElement('dependencies'))
 				Node dependency = dependencies.appendChild(document.createElement('dependency'))
 				Node groupId = dependency.appendChild(document.createElement('groupId'))
-				groupId.setTextContent('net.fabricmc')
+				groupId.setTextContent('org.quiltmc')
 				Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-				artifactId.setTextContent('fabric-loom')
+				artifactId.setTextContent('loom')
 				Node version = dependency.appendChild(document.createElement('version'))
 				version.setTextContent(baseVersion + '-SNAPSHOT')
 			})
@@ -263,6 +273,7 @@ publishing {
 	}
 	repositories {
 		maven {
+			name = 'release'
 			if (ENV.MAVEN_URL) {
 				url ENV.MAVEN_URL
 				credentials {
@@ -271,6 +282,26 @@ publishing {
 				}
 			}
 		}
+		maven {
+			name = "snapshot"
+			if (ENV.SNAPSHOTS_URL) {
+				url ENV.SNAPSHOTS_URL
+				credentials {
+					username ENV.SNAPSHOTS_USERNAME
+					password ENV.SNAPSHOTS_PASSWORD
+				}
+			}
+		}
+	}
+}
+
+tasks.withType(PublishToMavenRepository) {
+	onlyIf {
+		(repository == publishing.repositories.release && publication == publishing.publications.pluginMaven) ||
+				(repository == publishing.repositories.snapshot
+				&& (publication == publishing.publications.snapshot
+				|| publication == publishing.publications.snapshotPlugin)) ||
+				(repository == publishing.repositories.release && publication.name == "quiltLoomPluginMarkerMaven")
 	}
 }
 
diff --git a/gradle.properties b/gradle.properties
index 31014b156127886c44921900f480e68d4528da25..ee1d4c3083cdb729df09522e88a41351d332a875 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,5 +1,5 @@
-name = fabric-loom
-description = The Gradle plugin for Fabric
-url = https://github.com/FabricMC/fabric-loom
+name = loom
+description = The Gradle plugin for Quilt
+url = https://github.com/QuiltMC/quilt-loom
 
 kotlin.stdlib.default.dependency = false
\ No newline at end of file
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index 9c9c938e2e57b1d41c732014a2d6f5bcefd37e7b..5f54c3745bd0e89cb3df6b56e8b2a36fcfb9e60b 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -53,7 +53,7 @@ import net.fabricmc.loom.util.LibraryLocationLogger;
 public class LoomGradlePlugin implements BootstrappedPlugin {
 	public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
 	public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-	public static final String LOOM_VERSION = Objects.requireNonNullElse(LoomGradlePlugin.class.getPackage().getImplementationVersion(), "0.0.0+unknown");
+	public static final String LOOM_VERSION = Objects.requireNonNullElse(LoomGradlePlugin.class.getPackage().getImplementationVersion(), "0.12.0+unknown");
 
 	/**
 	 * An ordered list of setup job classes.
@@ -79,7 +79,7 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
 	}
 
 	public void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quilt Loom: " + LOOM_VERSION);
 		LibraryLocationLogger.logLibraryVersions();
 
 		// Apply default plugins
@@ -94,5 +94,7 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
 		for (Class<? extends Runnable> jobClass : SETUP_JOBS) {
 			project.getObjects().newInstance(jobClass).run();
 		}
+
+		project.apply(ImmutableMap.of("plugin", "io.github.juuxel.loom-vineflower"));
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java b/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java
index 29b4a526050a69cb5ccaa2d067fa5ecaa0347bf0..dc4895d24e908cf5d041ae71ef17dac40e6c86bf 100644
--- a/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java
@@ -67,7 +67,10 @@ public class LoomRepositoryPlugin implements Plugin<PluginAware> {
 			repo.setName("Fabric");
 			repo.setUrl(MirrorUtil.getFabricRepository(target));
 		});
-
+		repositories.maven(repo -> {
+			repo.setName("Quilt Releases");
+			repo.setUrl(MirrorUtil.getQuiltRepository(target));
+		});
 		MavenArtifactRepository mojangRepo = repositories.maven(repo -> {
 			repo.setName("Mojang");
 			repo.setUrl(MirrorUtil.getLibrariesBase(target));
diff --git a/src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java b/src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java
index 925f5a7cb85c89f9596dd3de078c6d61d384595f..fd1664dd7a391de74dc0befd1d57486bc7894c64 100644
--- a/src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java
+++ b/src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java
@@ -177,7 +177,7 @@ public final class IncludedJarFactory {
 				.replaceAll("\\.", "_")
 				.toLowerCase(Locale.ENGLISH);
 
-		// Fabric Loader can't handle modIds longer than 64 characters
+		// Loader can't handle modIds longer than 64 characters
 		if (modId.length() > 64) {
 			String hash = Hashing.sha256()
 					.hashString(modId, StandardCharsets.UTF_8)
diff --git a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
index 8930e596efbe20bbca62c77a2943eff428cb6554..d239a6ded6a179cb0bfb4c351bd95be631593b30 100644
--- a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
@@ -178,16 +178,16 @@ public abstract class CompileConfiguration implements Runnable {
 		final LoomGradleExtension extension = configContext.extension();
 
 		final boolean enableTransitiveAccessWideners = extension.getEnableTransitiveAccessWideners().get();
-		extension.addMinecraftJarProcessor(AccessWidenerJarProcessor.class, "fabric-loom:access-widener", enableTransitiveAccessWideners, extension.getAccessWidenerPath());
+		extension.addMinecraftJarProcessor(AccessWidenerJarProcessor.class, "quilt-loom:access-widener", enableTransitiveAccessWideners, extension.getAccessWidenerPath());
 
 		if (extension.getEnableModProvidedJavadoc().get()) {
-			extension.addMinecraftJarProcessor(ModJavadocProcessor.class, "fabric-loom:mod-javadoc");
+			extension.addMinecraftJarProcessor(ModJavadocProcessor.class, "quilt-loom:mod-javadoc");
 		}
 
 		final InterfaceInjectionExtensionAPI interfaceInjection = extension.getInterfaceInjection();
 
 		if (interfaceInjection.isEnabled()) {
-			extension.addMinecraftJarProcessor(InterfaceInjectionProcessor.class, "fabric-loom:interface-inject", interfaceInjection.getEnableDependencyInterfaceInjection().get());
+			extension.addMinecraftJarProcessor(InterfaceInjectionProcessor.class, "quilt-loom:interface-inject", interfaceInjection.getEnableDependencyInterfaceInjection().get());
 		}
 	}
 
diff --git a/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java b/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
index b41b39bf86c6bfcd8c600eb70d160d801e66f510..7d52086caedb771641e5355abddb3cb8f2345036 100644
--- a/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
+++ b/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
@@ -44,7 +44,7 @@ public class LoomDependencyManager {
 		sourceRemapper.remapAll();
 
 		if (extension.getInstallerData() == null) {
-			project.getLogger().warn("fabric-installer.json not found in dependencies!");
+			project.getLogger().warn("quilt_installer.json not found in dependencies!");
 		}
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
index 538c3018b085067e3d0e99d6fc8a0530b07cf6e9..9e4b9dbb1d8d837653fe4d1acebebfc22fed7bdb 100644
--- a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
@@ -68,7 +68,7 @@ public class SingleJarDecompileConfiguration extends DecompileConfiguration<Mapp
 			task.setDescription("Decompile minecraft using the default decompiler.");
 			task.setGroup(Constants.TaskGroup.FABRIC);
 
-			task.dependsOn(project.getTasks().named("genSourcesWithCfr"));
+			task.dependsOn(project.getTasks().named("genSourcesWithVineflower"));
 		});
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/configuration/decompile/SplitDecompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/decompile/SplitDecompileConfiguration.java
index 0cf8fbe389b957ca3d2024afd98b5a52e6a537b0..3d1a030cf2fbe7fbf57f834f362cb8aa93e22b12 100644
--- a/src/main/java/net/fabricmc/loom/configuration/decompile/SplitDecompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/decompile/SplitDecompileConfiguration.java
@@ -108,7 +108,7 @@ public final class SplitDecompileConfiguration extends DecompileConfiguration<Ma
 			task.setDescription("Decompile minecraft (%s) using the default decompiler.".formatted(name));
 			task.setGroup(Constants.TaskGroup.FABRIC);
 
-			task.dependsOn(project.getTasks().named("gen%sSourcesWithCfr".formatted(name)));
+			task.dependsOn(project.getTasks().named("gen%sSourcesWithVineflower".formatted(name)));
 		});
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
index 024a35c8843e8f866f3885a1414124854bfe97ef..7274912e12820e697991389626d5bda87ddd1162 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
@@ -42,7 +42,7 @@ import net.fabricmc.loom.util.FileSystemUtil;
 import net.fabricmc.loom.util.fmj.FabricModJsonFactory;
 
 public record ArtifactMetadata(boolean isFabricMod, RemapRequirements remapRequirements, @Nullable InstallerData installerData) {
-	private static final String INSTALLER_PATH = "fabric-installer.json";
+	private static final String INSTALLER_PATH = "quilt_installer.json";
 	private static final String MANIFEST_PATH = "META-INF/MANIFEST.MF";
 	private static final String MANIFEST_REMAP_KEY = "Fabric-Loom-Remap";
 
diff --git a/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java b/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
index d7f0c469b92b3acfb24a623306f8d63bf44eafbe..12f19648c39d6eec91d8bd09cdd3400aadac3ca3 100644
--- a/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
+++ b/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
@@ -50,22 +50,22 @@ public abstract class LoomFilesBaseImpl implements LoomFiles {
 
 	@Override
 	public File getUserCache() {
-		return createFile(getGradleUserHomeDir(), "caches" + File.separator + "fabric-loom");
+		return createFile(getGradleUserHomeDir(), "caches" + File.separator + "quilt-loom");
 	}
 
 	@Override
 	public File getRootProjectPersistentCache() {
-		return createFile(getRootDir(), ".gradle" + File.separator + "loom-cache");
+		return createFile(getRootDir(), ".gradle" + File.separator + "quilt-loom-cache");
 	}
 
 	@Override
 	public File getProjectPersistentCache() {
-		return createFile(getProjectDir(), ".gradle" + File.separator + "loom-cache");
+		return createFile(getProjectDir(), ".gradle" + File.separator + "quilt-loom-cache");
 	}
 
 	@Override
 	public File getProjectBuildCache() {
-		return createFile(getBuildDir(), "loom-cache");
+		return createFile(getBuildDir(), "quilt-loom-cache");
 	}
 
 	@Override
diff --git a/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java b/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
index c47746099e922ab35c6528bf6992a3b1ed063018..d0cce1fab554e97e7dfca0855068b82a4e601bbc 100644
--- a/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
+++ b/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
@@ -55,15 +55,17 @@ public abstract class GenerateDLIConfigTask extends AbstractLoomTask {
 		}
 
 		final LaunchConfig launchConfig = new LaunchConfig()
-				.property("fabric.development", "true")
-				.property("fabric.remapClasspathFile", getExtension().getFiles().getRemapClasspathFile().getAbsolutePath())
+				.property("loader.development", "true")
+				.property("loader.remapClasspathFile", getExtension().getFiles().getRemapClasspathFile().getAbsolutePath())
 				.property("log4j.configurationFile", getAllLog4JConfigFiles())
 				.property("log4j2.formatMsgNoLookups", "true")
 
 				.argument("client", "--assetIndex")
 				.argument("client", getExtension().getMinecraftProvider().getVersionInfo().assetIndex().fabricId(getExtension().getMinecraftProvider().minecraftVersion()))
 				.argument("client", "--assetsDir")
-				.argument("client", assetsDirectory.getAbsolutePath());
+				.argument("client", assetsDirectory.getAbsolutePath())
+				.argument("client", "--version")
+				.argument("client", "Quilt Loom");
 
 		if (versionInfo.hasNativesToExtract()) {
 			String nativesPath = getExtension().getFiles().getNativesDirectory(getProject()).getAbsolutePath();
@@ -74,12 +76,12 @@ public abstract class GenerateDLIConfigTask extends AbstractLoomTask {
 		}
 
 		if (getExtension().areEnvironmentSourceSetsSplit()) {
-			launchConfig.property("client", "fabric.gameJarPath.client", getGameJarPath("client"));
-			launchConfig.property("fabric.gameJarPath", getGameJarPath("common"));
+			launchConfig.property("client", "loader.gameJarPath.client", getGameJarPath("client"));
+			launchConfig.property("loader.gameJarPath", getGameJarPath("common"));
 		}
 
 		if (!getExtension().getMods().isEmpty()) {
-			launchConfig.property("fabric.classPathGroups", getClassPathGroups());
+			launchConfig.property("loader.classPathGroups", getClassPathGroups());
 		}
 
 		final boolean plainConsole = getProject().getGradle().getStartParameter().getConsoleOutput() == ConsoleOutput.Plain;
diff --git a/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java b/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
index b3f5456c1317acc23b903df9451b59683d677dcf..18341c50640d4e42f2391ae46c3d2be772c3a18f 100644
--- a/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
+++ b/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
@@ -88,7 +88,7 @@ public final class SourceRemapperService implements SharedService {
 		// Create a temp directory with all of the sources
 		if (!Files.isDirectory(source)) {
 			isSrcTmp = true;
-			srcPath = Files.createTempDirectory("fabric-loom-src");
+			srcPath = Files.createTempDirectory("quilt-loom-src");
 			ZipUtils.unpackAll(source, srcPath);
 		}
 
diff --git a/src/main/java/net/fabricmc/loom/util/Constants.java b/src/main/java/net/fabricmc/loom/util/Constants.java
index eebfeee5b4b4245aa867ee4a3c8a1bf4ffae081a..d12a938af780a8adf995da406ce7a4c2274b7980 100644
--- a/src/main/java/net/fabricmc/loom/util/Constants.java
+++ b/src/main/java/net/fabricmc/loom/util/Constants.java
@@ -32,6 +32,7 @@ public class Constants {
 	public static final String VERSION_MANIFESTS = "https://piston-meta.mojang.com/mc/game/version_manifest_v2.json";
 	public static final String EXPERIMENTAL_VERSIONS = "https://maven.fabricmc.net/net/minecraft/experimental_versions.json";
 	public static final String FABRIC_REPOSITORY = "https://maven.fabricmc.net/";
+	public static final String QUILT_REPOSITORY = "https://maven.quiltmc.org/repository/release/";
 
 	public static final int ASM_VERSION = Opcodes.ASM9;
 
diff --git a/src/main/java/net/fabricmc/loom/util/MirrorUtil.java b/src/main/java/net/fabricmc/loom/util/MirrorUtil.java
index 3350a985f73e403f086a2a988283cbe4644d0cc2..eef1cd42b1e53497012165e1580c6199600f2f04 100644
--- a/src/main/java/net/fabricmc/loom/util/MirrorUtil.java
+++ b/src/main/java/net/fabricmc/loom/util/MirrorUtil.java
@@ -66,4 +66,12 @@ public class MirrorUtil {
 
 		return Constants.FABRIC_REPOSITORY;
 	}
+
+	public static String getQuiltRepository(ExtensionAware aware) {
+		if (aware.getExtensions().getExtraProperties().has("loom_quilt_repository")) {
+			return String.valueOf(aware.getExtensions().getExtraProperties().get("loom_quilt_repository"));
+		}
+
+		return Constants.QUILT_REPOSITORY;
+	}
 }
diff --git a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
index 17fa8a52e63b8d17b7ae5e72402b47f4046f7e31..0ad332ebf7bf1dc6316ba3c32303a685c55e02a7 100644
--- a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
+++ b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
@@ -128,7 +128,7 @@ public class SourceRemapper {
 		if (!source.isDirectory()) {
 			// create tmp directory
 			isSrcTmp = true;
-			srcPath = Files.createTempDirectory("fabric-loom-src");
+			srcPath = Files.createTempDirectory("quilt-loom-src");
 			ZipUtils.unpackAll(source.toPath(), srcPath);
 		}
 
diff --git a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
index 7fd5d15941dddd97c2631c26b243b7f320d9337f..7d7122b5403049714202a7ead9be65cea04585ce 100644
--- a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
@@ -55,7 +55,7 @@ public final class GradleUtils {
 	}
 
 	public static boolean isLoomProject(Project project) {
-		return project.getPluginManager().hasPlugin("fabric-loom");
+		return project.getPluginManager().hasPlugin("org.quiltmc.loom");
 	}
 
 	public static Provider<Boolean> getBooleanPropertyProvider(Project project, String key) {
diff --git a/src/main/java/net/fabricmc/loom/util/service/BuildSharedServiceManager.java b/src/main/java/net/fabricmc/loom/util/service/BuildSharedServiceManager.java
index 6575611dea711ca1bb284a76132110404cdd95cb..b6f0ed542dc183f5782e300c57dce2c3ee3969b2 100644
--- a/src/main/java/net/fabricmc/loom/util/service/BuildSharedServiceManager.java
+++ b/src/main/java/net/fabricmc/loom/util/service/BuildSharedServiceManager.java
@@ -39,7 +39,7 @@ import org.slf4j.LoggerFactory;
 
 public abstract class BuildSharedServiceManager implements BuildService<BuildServiceParameters.None> {
 	private static final Logger LOGGER = LoggerFactory.getLogger(BuildSharedServiceManager.class);
-	private static final String NAME = "loom:sharedServiceManager";
+	private static final String NAME = "quilt-loom:sharedServiceManager";
 
 	private SharedServiceManager sharedServiceManager = new BuildSharedServiceManagerImpl();
 	private final AtomicInteger refCount = new AtomicInteger(0);
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
index 8b4de3243a929eb4073318b68c94ea6adb300438..0051c810071cc95fe89072e2bb8e5ade00c0c27c 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
@@ -45,7 +45,7 @@ class CustomManifestTest extends Specification implements GradleProjectTestTrait
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18_experimental-snapshot-1"
                     mappings "net.fabricmc:yarn:1.18_experimental-snapshot-1+build.2:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.11.6"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 		when:
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
index 5b2f0e105991b8493ca4473a0570cd7457156c07..ac4f40854f8e8cba0ff431292f32f063aa625527 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
@@ -46,6 +46,7 @@ class DecompileTest extends Specification implements GradleProjectTestTrait {
 
 		where:
 		decompiler 		| task								| version
+		'quiltflower'	| "genSourcesWithQuiltflower"		| PRE_RELEASE_GRADLE
 		'fernflower'	| "genSourcesWithFernFlower"		| PRE_RELEASE_GRADLE
 		'cfr' 			| "genSourcesWithCfr"				| PRE_RELEASE_GRADLE
 	}
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
index 7aa4d6dc560ebd986e1bc4ba7c01cc9f4e814dcd..58a0b22a2c68eda2e343b9b90a463551ecf53b03 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
@@ -42,7 +42,7 @@ class ExperimentalVersionsTest extends Specification implements GradleProjectTes
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18_experimental-snapshot-1"
                     mappings "net.fabricmc:yarn:1.18_experimental-snapshot-1+build.2:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.11.6"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
index 907de8f11f8e126ec3bba2664a01cfc29c6a5e74..b7a91e206add4ac4729286560872eab294162b14 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
@@ -64,7 +64,7 @@ class LegacyProjectTest extends Specification implements GradleProjectTestTrait
 						// No names
 					}
 
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
 			"""
 
@@ -102,7 +102,7 @@ class LegacyProjectTest extends Specification implements GradleProjectTestTrait
 						// No names
 					}
 
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
 			"""
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/LocalRuntimeTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/LocalRuntimeTest.groovy
index b357e20dfcb28deb80d5ecaa72f902af4ef10635..59635f441031c1d22ba0eb4f73bb683aef06b3a1 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/LocalRuntimeTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/LocalRuntimeTest.groovy
@@ -48,7 +48,7 @@ class LocalRuntimeTest extends Specification implements GradleProjectTestTrait {
 		result.task(":build").outcome == SUCCESS
 		def pomFile = new File(gradle.getProjectDir(), "build/publications/mavenJava/pom-default.xml")
 		def pom = pomFile.text
-		!pom.contains("fabric-api")
+		// !pom.contains("fabric-api") Quilt: this doesn't work because Quilt Loader specifies a `fabric-api` exclusion
 		!pom.contains("enigma")
 
 		where:
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
index a7265d4e57518fac94846b67f84688a73e483a14..31a052b1bce8a21003827e98be59f9d416ccbf93 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
@@ -46,7 +46,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -73,7 +73,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -100,7 +100,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -127,7 +127,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
index 770a7b2302c37f28d3d0cef2fbdcebf469eefb38..6c453b1ddbd4e1916daa08e96beb1d3bcc42c11a 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
@@ -88,7 +88,7 @@ class RunConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
index d434f05d759c48cd8e93dc5f7238a7e961f3ee53..3ee7751f63840fd2ff5847c03678d9584fa38bfb 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
@@ -51,7 +51,7 @@ class SimpleProjectTest extends Specification implements GradleProjectTestTrait
 		def serverResult = server.run()
 		then:
 		result.task(":build").outcome == SUCCESS
-		gradle.getOutputZipEntry("fabric-example-mod-1.0.0.jar", "META-INF/MANIFEST.MF").contains("Fabric-Loom-Version: 0.0.0+unknown")
+		gradle.getOutputZipEntry("fabric-example-mod-1.0.0.jar", "META-INF/MANIFEST.MF").contains("Fabric-Loom-Version: 0.12.0+unknown")
 		gradle.getOutputZipEntry("fabric-example-mod-1.0.0-sources.jar", "net/fabricmc/example/mixin/ExampleMixin.java").contains("class_442") // Very basic test to ensure sources got remapped
 
 		serverResult.successful()
diff --git a/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy b/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
index a741f095beaa9510e4c723b7d8fc6079a91790db..0ee6af97f298a13b35ea6c9f794592fc6ec8fdce 100644
--- a/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
@@ -237,7 +237,7 @@ trait GradleProjectTestTrait {
 		}
 
 		File getGeneratedSources(String mappings, String jarType = "merged") {
-			return new File(getGradleHomeDir(), "caches/fabric-loom/minecraftMaven/net/minecraft/minecraft-${jarType}/${mappings}/minecraft-${jarType}-${mappings}-sources.jar")
+			return new File(getGradleHomeDir(), "caches/quilt-loom/minecraftMaven/net/minecraft/minecraft-${jarType}/${mappings}/minecraft-${jarType}-${mappings}-sources.jar")
 		}
 
 		File getGeneratedLocalSources(String mappings) {
diff --git a/src/test/resources/projects/accesswidener/build.gradle b/src/test/resources/projects/accesswidener/build.gradle
index af697460cc9366f08ac6084c7d5dba1244426dca..b02f00aab16a6e4a851f4389c917bad3b8a579cf 100644
--- a/src/test/resources/projects/accesswidener/build.gradle
+++ b/src/test/resources/projects/accesswidener/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/accesswidener/gradle.properties b/src/test/resources/projects/accesswidener/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/accesswidener/gradle.properties
+++ b/src/test/resources/projects/accesswidener/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index 993ac667071e4e3e2b1a8301121e817950663d61..17961497657cb80f2ce1cb2463f1a28c11e85c5d 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -1,11 +1,11 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 dependencies {
 	minecraft "com.mojang:minecraft:21w38a"
 	mappings "net.fabricmc:yarn:21w38a+build.11:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.7"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
 
 loom {
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
index 1ad70fd35b62504694311c46182f7004762e2213..5b2ca94125cd456b7073889c66b205f3f61fe161 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties b/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
index 40e9e5717128a126a91dcfcc0b6fd62fb9100792..523f9a2e212924f27dd2c6f66305b761c3e0c247 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.6
-	loader_version=0.11.3
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
index 11f94b3c2b6defeca5e16769192b0dee0d73b547..85bbccb051c0a92eefeff7bae775d3e233c96df4 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties b/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
index 40e9e5717128a126a91dcfcc0b6fd62fb9100792..523f9a2e212924f27dd2c6f66305b761c3e0c247 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.6
-	loader_version=0.11.3
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
index 2c075f03ad35f9065fc5d9e501a236ede735a016..365f417912dd238bf5dbf4ace67ec36ef5038bf9 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
@@ -1,11 +1,11 @@
 pluginManagement {
 	plugins {
-		id 'fabric-loom'
+		id 'org.quiltmc.loom'
 	}
 }
 
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 dependencyResolutionManagement {
diff --git a/src/test/resources/projects/interfaceInjection/build.gradle b/src/test/resources/projects/interfaceInjection/build.gradle
index e631fd01bbce95d0eb6c494034cfe5fa5ec781f5..098666f23fe51b71cc4720b02ed64c0306894262 100644
--- a/src/test/resources/projects/interfaceInjection/build.gradle
+++ b/src/test/resources/projects/interfaceInjection/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -11,7 +11,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	// I-faces will still be applied as the jar is on both configurations.
 	modCompileOnly files("dummy.jar")
diff --git a/src/test/resources/projects/java16/build.gradle b/src/test/resources/projects/java16/build.gradle
index 018aeb560dad93d12ab7646fb1cd86c14cbc866d..dc6761d099c47dfcd2f009ab336b76e518c9fe9e 100644
--- a/src/test/resources/projects/java16/build.gradle
+++ b/src/test/resources/projects/java16/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 }
 
diff --git a/src/test/resources/projects/java16/gradle.properties b/src/test/resources/projects/java16/gradle.properties
index cdc647ce754f229ba0daa586be4031a0b734251e..681c2d0af7e67f717c0f28bda66a911da8b2490e 100644
--- a/src/test/resources/projects/java16/gradle.properties
+++ b/src/test/resources/projects/java16/gradle.properties
@@ -2,7 +2,7 @@ org.gradle.jvmargs=-Xmx1G
 
 minecraft_version=1.18-pre1
 yarn_mappings=1.18-pre1+build.2
-loader_version=0.12.5
+loader_version=0.16.0-beta.16
 fabric_version=0.42.1+1.18
 
 mod_version = 1.0.0
diff --git a/src/test/resources/projects/kotlin/build.gradle.kts b/src/test/resources/projects/kotlin/build.gradle.kts
index 7f2e0f5d28bd402223319574cf804cc7ddeabc62..8175dce88d1b8f226dbb4b2db58744cfd458870c 100644
--- a/src/test/resources/projects/kotlin/build.gradle.kts
+++ b/src/test/resources/projects/kotlin/build.gradle.kts
@@ -3,7 +3,7 @@ import java.util.Properties
 plugins {
 	kotlin("jvm") version "1.7.22"
 	kotlin("plugin.serialization") version "1.7.22"
-	id("fabric-loom")
+	id("org.quiltmc.loom")
 	`maven-publish`
 }
 
@@ -18,7 +18,7 @@ version = "0.0.1"
 dependencies {
 	minecraft(group = "com.mojang", name = "minecraft", version = "1.16.5")
 	mappings(group = "net.fabricmc", name = "yarn", version = "1.16.5+build.5", classifier = "v2")
-	modImplementation("net.fabricmc:fabric-loader:0.12.12")
+	modImplementation("org.quiltmc:quilt-loader:0.16.1")
 	modImplementation(group = "net.fabricmc", name = "fabric-language-kotlin", version = "1.8.7+kotlin.1.7.22")
 }
 
diff --git a/src/test/resources/projects/legacy/build.gradle b/src/test/resources/projects/legacy/build.gradle
index 13e303a919faea188a009bd2152a78dd879777ed..b38742138392a427ce8376bbd9036341b0adc106 100644
--- a/src/test/resources/projects/legacy/build.gradle
+++ b/src/test/resources/projects/legacy/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 version = project.mod_version
@@ -8,7 +8,7 @@ group = project.maven_group
 dependencies {
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 }
 
diff --git a/src/test/resources/projects/legacy/gradle.properties b/src/test/resources/projects/legacy/gradle.properties
index 440dabe5b015628affe10c53ea2a8fcb977e9409..aba0090df90da5eae231db6dc1c61040a7d3d029 100644
--- a/src/test/resources/projects/legacy/gradle.properties
+++ b/src/test/resources/projects/legacy/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.14.4
 	yarn_mappings=1.14.4+build.18
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/localFileDependency/build.gradle b/src/test/resources/projects/localFileDependency/build.gradle
index 289b506359c6dff38067b1494ac4835a3bfd082d..45cf7c7f9768d2023c5115c3cd780ab9e993d91e 100644
--- a/src/test/resources/projects/localFileDependency/build.gradle
+++ b/src/test/resources/projects/localFileDependency/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -21,7 +21,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/localFileDependency/gradle.properties b/src/test/resources/projects/localFileDependency/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/localFileDependency/gradle.properties
+++ b/src/test/resources/projects/localFileDependency/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/localRuntime/build.gradle b/src/test/resources/projects/localRuntime/build.gradle
index a4f9a87821f907df4e36bcc10ca4622a74b4ffdc..c61947cd0a2fe9539ebc35a080f580c2c0760f7c 100644
--- a/src/test/resources/projects/localRuntime/build.gradle
+++ b/src/test/resources/projects/localRuntime/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modLocalRuntime "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/localRuntime/gradle.properties b/src/test/resources/projects/localRuntime/gradle.properties
index 845c356072dbb73ddb31b97826e8cb4fbaf138e0..c01c88366c6038008ee1d2f668b63bc2de2e138a 100644
--- a/src/test/resources/projects/localRuntime/gradle.properties
+++ b/src/test/resources/projects/localRuntime/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	maven_group = com.example
diff --git a/src/test/resources/projects/maven/build.gradle b/src/test/resources/projects/maven/build.gradle
index f7069b886efbb520321cac93e2758c4ffccd28b7..673594253d86a9338ac42b533ec6d2e1617b2228 100644
--- a/src/test/resources/projects/maven/build.gradle
+++ b/src/test/resources/projects/maven/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -21,7 +21,7 @@ configurations {
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	testConfig System.getProperty("loom.test.resolve")
 }
diff --git a/src/test/resources/projects/mavenLibrary/build.gradle b/src/test/resources/projects/mavenLibrary/build.gradle
index 95703b93c317eb35390fd945fdacbc46ff12deff..d954f2c708ee388b5334631a70bd83a5654d14e6 100644
--- a/src/test/resources/projects/mavenLibrary/build.gradle
+++ b/src/test/resources/projects/mavenLibrary/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -9,7 +9,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 	modCompileOnlyApi fabricApi.module('fabric-biome-api-v1', '0.42.0+1.16')
 	modRuntimeOnly fabricApi.module('fabric-command-api-v1', '0.42.0+1.16')
 }
diff --git a/src/test/resources/projects/minimalBase/build.gradle b/src/test/resources/projects/minimalBase/build.gradle
index f3dc73941bc7331e8292466b924f4a769ae55a01..f6801db584e32d4ed189ac09b4253b28dbf8bdf5 100644
--- a/src/test/resources/projects/minimalBase/build.gradle
+++ b/src/test/resources/projects/minimalBase/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApAutoRefmap/build.gradle b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
index 058aef168b3bcb720935aac393c36b6c52a3538b..ae1ad5d8f6604dfc279e03054ed3dbcacfac9df9 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/build.gradle
+++ b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'com.github.johnrengelman.shadow' version '7.0.0'
 	id 'maven-publish'
 }
@@ -19,7 +19,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mixinApAutoRefmap/gradle.properties b/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
index 60cb9a4f2c0b4fca94a187862c8a28afbb2a00cf..0925278e52f63aa39b68cf7aa6b5107541317c20 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
+++ b/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/versions.html
 	minecraft_version=1.17
 	yarn_mappings=1.17+build.13
-	loader_version=0.11.6
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index 68b48e9c25d51cd96e7149243e911949262b5917..589180d6b540efa76a78445d688b14a32577dbd3 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mixinApSimple/gradle.properties b/src/test/resources/projects/mixinApSimple/gradle.properties
index 60cb9a4f2c0b4fca94a187862c8a28afbb2a00cf..0925278e52f63aa39b68cf7aa6b5107541317c20 100644
--- a/src/test/resources/projects/mixinApSimple/gradle.properties
+++ b/src/test/resources/projects/mixinApSimple/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/versions.html
 	minecraft_version=1.17
 	yarn_mappings=1.17+build.13
-	loader_version=0.11.6
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/modJavadoc/build.gradle b/src/test/resources/projects/modJavadoc/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..bbbd52e6335960d9ae21a6c1358f30910be88d3c 100644
--- a/src/test/resources/projects/modJavadoc/build.gradle
+++ b/src/test/resources/projects/modJavadoc/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -11,7 +11,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.16"
 
 	modImplementation files("dummy.jar")
 }
diff --git a/src/test/resources/projects/mojangMappings/build.gradle b/src/test/resources/projects/mojangMappings/build.gradle
index f58c485386bbf686967a3020c2326996da62441d..30f47e97d864d1b7301cd91f86cd39f410e57fec 100644
--- a/src/test/resources/projects/mojangMappings/build.gradle
+++ b/src/test/resources/projects/mojangMappings/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings loom.officialMojangMappings()
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mojangMappings/gradle.properties b/src/test/resources/projects/mojangMappings/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/mojangMappings/gradle.properties
+++ b/src/test/resources/projects/mojangMappings/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/multi-mc-versions/build.gradle b/src/test/resources/projects/multi-mc-versions/build.gradle
index faa804c3e2416c4e6f25842f74436c4f215141a3..414b484bbf0aeb398d99c011a635987fe9ea2265 100644
--- a/src/test/resources/projects/multi-mc-versions/build.gradle
+++ b/src/test/resources/projects/multi-mc-versions/build.gradle
@@ -2,7 +2,7 @@ import groovy.json.JsonSlurper
 
 plugins {
 	id "java"
-	id 'fabric-loom' apply false
+	id 'org.quiltmc.loom' apply false
 }
 
 allprojects {
@@ -19,7 +19,7 @@ def getMappingVersion(String mcVersion) {
 }
 
 subprojects {
-	apply plugin: "fabric-loom"
+	apply plugin: "org.quiltmc.loom"
 
 	base {
 		archivesName = rootProject.name + "-" + project.name
diff --git a/src/test/resources/projects/multiproject/build.gradle b/src/test/resources/projects/multiproject/build.gradle
index ae3e637a0b52af9f00d9742c331f0cb943aaf9d4..af8bc552dda1e9e53af5d2a297c4cb444481e5ab 100644
--- a/src/test/resources/projects/multiproject/build.gradle
+++ b/src/test/resources/projects/multiproject/build.gradle
@@ -1,11 +1,11 @@
 plugins {
 	id "java"
 	id "java-library"
-	id 'fabric-loom' apply false
+	id 'org.quiltmc.loom' apply false
 }
 
 allprojects {
-	apply plugin: "fabric-loom"
+	apply plugin: "org.quiltmc.loom"
 
 	version = "1.0.0"
 
@@ -25,7 +25,7 @@ allprojects {
 		// To change the versions see the gradle.properties file
 		minecraft "com.mojang:minecraft:1.16.5"
 		mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-		modImplementation "net.fabricmc:fabric-loader:0.11.2"
+		modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 		modImplementation "net.fabricmc.fabric-api:fabric-api:0.31.0+1.16"
 	}
diff --git a/src/test/resources/projects/multiproject/gradle.properties b/src/test/resources/projects/multiproject/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/multiproject/gradle.properties
+++ b/src/test/resources/projects/multiproject/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/parchment/build.gradle b/src/test/resources/projects/parchment/build.gradle
index 5fee81c5d1e07de0dfcf615795ec818a8e4705e2..76ec0e30196bfae2f5b92b2e8b722256d980da0e 100644
--- a/src/test/resources/projects/parchment/build.gradle
+++ b/src/test/resources/projects/parchment/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 repositories {
@@ -18,5 +18,5 @@ dependencies {
 		}
 	}
 
-	modImplementation libs.fabricLoader
+	modImplementation libs.loader
 }
diff --git a/src/test/resources/projects/parchment/gradle/libs.versions.toml b/src/test/resources/projects/parchment/gradle/libs.versions.toml
index 40c22f786bae76ab2365e2a337d8a25486a5b9e3..cd3bac3e45d8aceb1acba8ace079bfce926b4f64 100644
--- a/src/test/resources/projects/parchment/gradle/libs.versions.toml
+++ b/src/test/resources/projects/parchment/gradle/libs.versions.toml
@@ -1,9 +1,9 @@
 [versions]
 minecraft = "1.16.5"
-fabricLoader = "0.11.3"
+loader = "0.16.0-beta.5"
 parchment = "2021.10.10"
 
 [libraries]
 minecraft = { module = "com.mojang:minecraft", version.ref = "minecraft" }
-fabricLoader = { module = "net.fabricmc:fabric-loader", version.ref = "fabricLoader" }
+loader = { module = "org.quiltmc:quilt-loader", version.ref = "loader" }
 parchment = { module = "org.parchmentmc.data:parchment-1.16.5", version.ref = "parchment" }
\ No newline at end of file
diff --git a/src/test/resources/projects/reproducible/build.gradle b/src/test/resources/projects/reproducible/build.gradle
index 08c393f887b92c81a02841ad72ea90a8a0f1d0ab..c92ac92eae001f50f29a52bfc4a5e08b4652c306 100644
--- a/src/test/resources/projects/reproducible/build.gradle
+++ b/src/test/resources/projects/reproducible/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -15,7 +15,7 @@ loom {
 dependencies {
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 }
 
 base {
diff --git a/src/test/resources/projects/reproducible/gradle.properties b/src/test/resources/projects/reproducible/gradle.properties
index 3dc7831a9262fea0e3779896665bdc7a8c979f16..ce2badeb0f1e4860637304a0ba79f59d5fe0473e 100644
--- a/src/test/resources/projects/reproducible/gradle.properties
+++ b/src/test/resources/projects/reproducible/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index 5e504e20d2110896de2d182f0524be6cfe065411..20742e6a72340b04b4cf22ba690f86aea1856d0a 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 loom {
@@ -37,7 +37,7 @@ version = "1.0.0"
 dependencies {
 	minecraft "com.mojang:minecraft:1.18.1"
 	mappings "net.fabricmc:yarn:1.18.1+build.12:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.12.12"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
 
 base {
diff --git a/src/test/resources/projects/signed/build.gradle b/src/test/resources/projects/signed/build.gradle
index 8a5445b05b1bd8953409adbb4ea0cd31fd4fcc56..cf303f05d7d62cff2fa6e2bb446edcaecc28d8b8 100644
--- a/src/test/resources/projects/signed/build.gradle
+++ b/src/test/resources/projects/signed/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 	id 'signing'
 }
@@ -10,7 +10,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
 
 base {
diff --git a/src/test/resources/projects/simple/build.gradle b/src/test/resources/projects/simple/build.gradle
index ed7d9e49fa03821166e8f920e6497e2d66e42f82..22bc029e0b2744c6464729db0f16110e6268f36f 100644
--- a/src/test/resources/projects/simple/build.gradle
+++ b/src/test/resources/projects/simple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/simple/gradle.properties b/src/test/resources/projects/simple/gradle.properties
index 845c356072dbb73ddb31b97826e8cb4fbaf138e0..c01c88366c6038008ee1d2f668b63bc2de2e138a 100644
--- a/src/test/resources/projects/simple/gradle.properties
+++ b/src/test/resources/projects/simple/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	maven_group = com.example
diff --git a/src/test/resources/projects/splitSources/build.gradle b/src/test/resources/projects/splitSources/build.gradle
index baa2ebc18bdda779888e0620488ad0e03e663c10..b17193f0900bca32e7197dbf527e8afa2baa101e 100644
--- a/src/test/resources/projects/splitSources/build.gradle
+++ b/src/test/resources/projects/splitSources/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.12.local'
+	id 'org.quiltmc.loom' version '0.12.local'
 	id 'maven-publish'
 }
 
@@ -24,7 +24,7 @@ loom {
 dependencies {
 	minecraft "com.mojang:minecraft:1.18.2"
 	mappings "net.fabricmc:yarn:1.18.2+build.1:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.13.3"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	modImplementation "net.fabricmc.fabric-api:fabric-api:0.47.8+1.18.2"
 
diff --git a/src/test/resources/projects/transitiveAccesswidener/build.gradle b/src/test/resources/projects/transitiveAccesswidener/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..8eb07086160a68b46c98d8121ca8945851e59bee 100644
--- a/src/test/resources/projects/transitiveAccesswidener/build.gradle
+++ b/src/test/resources/projects/transitiveAccesswidener/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -11,7 +11,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	modImplementation files("dummy.jar")
 }
diff --git a/src/test/resources/projects/unpick/build.gradle b/src/test/resources/projects/unpick/build.gradle
index 4b79a1b01f6d59fea5174e912c213a1ee597327f..3d4edf7fe25337ea5db31367ec65cf207ac98f9b 100644
--- a/src/test/resources/projects/unpick/build.gradle
+++ b/src/test/resources/projects/unpick/build.gradle
@@ -1,9 +1,9 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 dependencies {
 	minecraft "com.mojang:minecraft:21w13a"
 	mappings "net.fabricmc:yarn:21w13a+build.30:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.3"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
